### PR TITLE
Update Envoy to the same version used by Istio 0.8.0 pre-release

### PR DESCRIPTION
**Pull request by rlenglet**
Signed-off-by: Romain Lenglet <romain@covalent.io>

### CODE REVIEW

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
The CI build timed out:
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/2623/
For instance, Runtime tests took 1h23, and K8s tests took 1h30 before getting killed.

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
test-me-please

**Review by jrajahalme — APPROVED** [review]

**Comment by jrajahalme** [comment]
Needs a rebase due to the segfault fix.

**Comment by jrajahalme** [comment]
While you are at it, this branch has a bit cleaner replacements for the http_connection_manager.h in cilium_l7policy.h & .cc: https://github.com/cilium/cilium/tree/pr/envoy-rebase-2018-04-21

**Comment by jrajahalme** [comment]
Should also rebuild the v2 API .pb.go files and include changes in them in the PR.

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
I pushed a new Docker image `quay.io/cilium/cilium-builder:2018-04-23` and I'm using it in this PR.

@eloycoto I also updated the image used in `test/docker-compose.yml`. It wasn't updated to use the quay.io images and was still using `cilium/cilium-builder:2018-03-29`.

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
CI build failure:
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/2647/execution/node/60/log/
```
19:55:27 RuntimeValidatedPolicyEnforcement
19:55:27 /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:280
19:55:27   Policy Enforcement Always
19:55:27   /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:280
19:55:27     Always to Never without policy [It]
19:55:27     /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:312
19:55:27 
19:55:27     Expected
19:55:27         <bool>: false
19:55:27     to be true
19:55:27 
19:55:27     /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/runtime/Policies.go:296
```

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
Green build with the updated cilium-builder image:
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/2651/

**Comment by rlenglet** [comment]
Waiting on a green `packer-ci-build` build for this branch, then will update with the updated Vagrant VM box version:
https://jenkins.cilium.io/job/cilium/job/packer-ci-build/job/master/57/

**Comment by rlenglet** [comment]
Got a green build of the Vagrant box (version 57):
https://jenkins.cilium.io/job/cilium/job/packer-ci-build/job/master/57/
Updated the `Vagrantfile`s in this PR.

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
Manually tested that the dev Vagrant box v57 comes up fine with this branch.

**Comment by rlenglet** [comment]
Green CI build:
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/2655/

File: Makefile [hunk 1]
```diff
@@ -131,7 +131,8 @@ docker-image-runtime:
 	cd contrib/packaging/docker && docker build -t "cilium/cilium-runtime:$(UTC_DATE)" -f Dockerfile.runtime .
 
 docker-image-builder:
-	cd envoy && docker build -t "cilium/cilium-builder:$(UTC_DATE)" .
+	cd envoy && docker build -t "quay.io/cilium/cilium-builder:$(UTC_DATE)" .
+	@echo "docker push quay.io/cilium/cilium-builder:$(UTC_DATE)"
```
Comment (raybejjani) [inline Makefile]: Is this intended as an instruction? It might look like an actual line being executed.

**Review by raybejjani — APPROVED** [review]

**Review by eloycoto — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (rlenglet) [reply Makefile]: Yes it’s an instruction. I followed the same pattern as the other target that builds a Docker image.

**Comment by rlenglet** [comment]
Rebased.

**Comment by rlenglet** [comment]
test-me-please

**Pull request merged**